demos/snake/README.md

```diff
@@ -0,0 +1,174 @@
+
+       Renderer renderer(kScreenWidth, kScreenHeight, kGridWidth,  kGridHeight);
+       Controller controller;
+     -   Game game(kGridWidth, kGridHeight);
+     -   game.Run(controller, renderer, kMsPerFrame);
+     -   std::cout << "Game has terminated successfully!\n";
+     -   std::cout << "Score: " << game.GetScore() << "\n";
+     -   std::cout << "Size: " << game.GetSize() << "\n";
+
+     + Game *game;
+
+     + void mainLoopWrapper(){
+     +   game->Run(controller, renderer, kMsPerFrame);
+     + }
+
+     + int main() {
+     +   Game gameIns(kGridWidth, kGridHeight);
+     +   game = &gameIns;
+     +   emscripten_set_main_loop(mainLoopWrapper, 0, 1);
+         return 0;
+       }
+     ```
+
+   - Select the file `src -> game.cpp` from the `Project Explorer` sidebar and modify the code as
+     below. Then click the `SAVE` button to save the changes.
+
+     ```diff
+       // src/game.cpp
+     + #include <emscripten.h>
+
+       void Game::Run(Controller const &controller, Renderer &renderer,
+                     std::size_t target_frame_duration) {
+           Uint32 title_timestamp = SDL_GetTicks();
+           Uint32 frame_start;
+           Uint32 frame_end;
+           Uint32 frame_duration;
+           int frame_count = 0;
+           bool running = true;
+           int frame_count = 0;
+           bool running = true;
+     -    while (running) {
+             ...
+     -    }
+           ...
+       }
+
+       void Game::Update() {
+     -   if (!snake.alive) return;
+     +   if (!snake.alive){
+     +       // Use JS alert function to display the Score and Size when snake is died.
+     +       EM_ASM({
+     +           var message = 'Game Over...\nYour Score: ' +  $0 + '\nSize: '+ +$1;
+     +           alert(message);
+     +       }, score, snake.size);
+     +       emscripten_cancel_main_loop();
+     +       return;
+     +   }
+         ...
+       }
+     ```
+
+   - Select the file `CMakeLists.txt` from the `Project Explorer` sidebar and modify the code as
+     below. Then click the `SAVE` button to save the changes.
+
+     ```diff
+       // CMakeLists.txt
+       project(SDL2Test)
+     + # to generate html target
+     + set(CMAKE_EXECUTABLE_SUFFIX ".html")
+     ```
+
+     > NOTE. _The demo is using the `html` generated by emscripten, generally `html` target could be
+     > generated with `emcc/em++ -o output.html`, which is not suitable approach when integrating
+     > with `emcmake` and `emmake`. Following suggestion is one approach to build `html` target, you
+     > could find the other ways that work._
+
+   - Click `Build` in the top navigation panel to route us back to the `Build` page.
+
+7. Receive the recipe for errors not handled by Webinzier
+
+   > `The html target file is not compatible with build options "-sMODULARIZE -sEXPORT_NAME=Module_"`
+
+8. **`Config page`** - Click `Options` in the left navigation bar and click the switch of
+   `Modularize JS output` to disable
```

Comment:
We can create an advisor to detect such error and apply automatic action using `ConfigOptionChangeAction`.